.github/workflows/doc.yml

```diff
@@ -18,13 +24,57 @@ jobs:
           profile: minimal
           override: true
 
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
       - name: Build docs
         env:
           RUSTDOCFLAGS: --cfg docsrs -D broken_intra_doc_links
         run: |
           cargo doc --workspace --no-deps --features=permission-calculator
           cargo doc -p twilight-util --no-deps --all-features
 
+      - name: Cache docs build
+        uses: actions/cache@v2
+        id: restore-docs
+        with:
+          path: ./*
+          key: ${{ github.sha }}
+
+  deploy-docs:
+    name: Deploy docs
+    runs-on: ubuntu-latest
+    needs: build-docs
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Download cached docs
+        uses: actions/cache@v2
+        id: restore-docs
+        with:
+          path: ./*
+          key: ${{ github.sha }}
```

Comment:
moving this under the same job but making the steps conditional simplifies the action a lot and removes the cache hack

Reply:
It was like that before, however imo having an `if` on each step makes the actions uglier than putting it in one step, if the general consensus is to be in one job I can revert it however

Reply:
i'd like to avoid caching with two different methods (actions/cache or Swatinem) if it's unnecessary

Reply:
Okay, Swatinem cache will be removed and I will reads the if statements on each line